static/app/views/releases/detail/overview/sidebar/thresholdStatuses.tsx

```diff
@@ -0,0 +1,114 @@
+import {useCallback, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Client} from 'sentry/api';
+import * as SidebarSection from 'sentry/components/sidebarSection';
+import {IconCheckmark, IconWarning} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {Organization, Release, ReleaseProject} from 'sentry/types';
+import {getExactDuration} from 'sentry/utils/formatters';
+import useApi from 'sentry/utils/useApi';
+
+import {
+  ReleaseDetailsTable,
+  ReleaseDetailsTableRow,
+} from '../../../components/releaseDetailsSideTable';
+import {fetchThresholdStatuses} from '../../../utils/fetchThresholdStatus';
+import {ThresholdStatus, ThresholdStatusesQuery} from '../../../utils/types';
+
+type Props = {
+  organization: Organization;
+  project: Required<ReleaseProject>;
+  release: Release;
+  selectedEnvs: string[];
+};
+
+function ThresholdStatuses({project, release, organization, selectedEnvs}: Props) {
+  const api: Client = useApi();
+  const [thresholdStatuses, setThresholdStatuses] = useState<ThresholdStatus[]>([]);
+  const fetchThresholdStatusCallback = useCallback(async () => {
+    const fuzzSec = 30;
+    const start = new Date(new Date(release.dateCreated).getTime() - fuzzSec * 1000);
+    const end = new Date(new Date(release.dateCreated).getTime() + fuzzSec * 1000);
+
+    const releaseVersion: string = release.version;
+
+    const query: ThresholdStatusesQuery = {
+      start: start.toISOString(),
+      end: end.toISOString(),
+      release: [releaseVersion],
+      project: [project.slug],
+    };
+    if (selectedEnvs.length) {
+      query.environment = selectedEnvs;
+    }
+    return await fetchThresholdStatuses(organization, api, query);
+  }, [release, project, selectedEnvs, organization, api]);
+
+  useEffect(() => {
+    fetchThresholdStatusCallback().then(thresholds => {
+      const list = thresholds[`${project.slug}-${release.version}`];
+      const lastDeploy = release.lastDeploy;
+      const filtered = list?.filter(status => {
+        return status.environment?.name === lastDeploy?.environment;
+      });
```

Comment:
should the environment be passed into the query rather than filtering for it here?

Reply:
Actually good call
this could cause a confusing discrepancy between the env select and the lastDeploy.

I'll remove the last deploy env filter